src/facets/MiningPassFacet.sol

```diff
@@ -28,7 +28,7 @@ contract MiningPassFacet {
         uint256 _currentSeasonId = s.currentSeasonId;
         UserData storage _userData = s.usersData[_currentSeasonId][msg.sender];
         // check _tier is not 0
-        if (_tier == 0 || _tier <= _userData.miningPassTier) {
+        if (_tier == 0 || _tier <= _userData.miningPassTier || _tier > s.miningPassMaxTier) {
```

Comment:
```suggestion
        if (_tier == 0 || _tier <= _userData.miningPassTier || _tier > s.miningPassTierToFee[_tier] == 0) {
```

I think this would work too, and we avoid having another state variable - less gas ⛽

Reply:
sorry, is not actually less gas because we would still be accessing the state variable, but could be optimized